src/Swarm/Game/Value.hs

```diff
@@ -80,8 +84,17 @@ data Value where
   VDelay :: Term -> Env -> Value
   -- | A reference to a memory cell in the store.
   VRef :: Int -> Value
+  -- | A computation running in another thread.
+  VAsync :: Term -> GAsync Value -> Value
   deriving (Eq, Show)
 
+-- | Type for asynchronous computation which may raise an Exn.
+newtype GAsync a = GAsync {asyncComp :: Async.Async (Either Exn a)}
```

Comment:
Do we really need the `GAsync` type?  What's the benefit of defining `GAsync` as opposed to inlining its definition directly into `VAsync` (i.e. `VAsync :: Term -> Async.Async (Either Exn Value) -> Value`)?  It kind of feels like unnecessary indirection to me.